aiida_quantumespresso/calculations/__init__.py

```diff
@@ -285,10 +288,16 @@ def _generate_PWCPinputdata(cls, parameters, settings, pseudos, structure, kpoin
 
         # ============ I prepare the input site data =============
         # ------------ CELL_PARAMETERS -----------
-        cell_parameters_card = 'CELL_PARAMETERS angstrom\n'
-        for vector in structure.cell:
-            cell_parameters_card += ('{0:18.10f} {1:18.10f} {2:18.10f}'
-                                     '\n'.format(*vector))
+
+        # Specify cell parameters only if 'ibrav' is either zero or
+        # unspecified.
+        if input_params.get('SYSTEM', {}).get('ibrav', 0) == 0:
```

Comment:
maybe define a `DEFAULT_IBRAV = 0` in the class and use that in this line as well as in line 460. Not likely that we will change the default, but it does make it a bit clearer what this `0` is.

Reply:
Ah wait.. this logic actually works only if the default is zero - because for non-zero `ibrav` you can't specify the cell in the same way.

I agree that we want to make it clearer what this "magic zero" is, but I think just naming it `DEFAULT_IBRAV` is also dangerous: Simply changing the default will break the code.

Reply:
Hmm, I guess keeping the right-hand-side zero hard-coded and making the left one a variable works.

Reply:
good point. Just assigning it to variable to make its meaning clear is enough